File: dart/acronym/lib/acronym.dart
// Convert a phrase to its acronym.
// https://exercism.org/tracks/dart/exercises/acronym

import "dart:math";

class Acronym {
  String abbreviate(String input_text) => input_text
      .toUpperCase()
      .replaceAll(RegExp(r"[^- A-Z]"), "")
      .split(RegExp(r"[- ]"))
      .map((word) => word.substring(0, min(1, word.length)))
      .join();
}

File: ruby/acronym/acronym.rb
# frozen_string_literal: false

# https://exercism.org/tracks/ruby/exercises/acronym
# Acronym exercise
module Acronym
  def self.abbreviate(input)
    text = input.chomp.upcase

    first = true
    runes = text.chars.select do |rune|
      if first && ('A'..'Z').include?(rune)
        first = false
        true
      elsif [' ', '_', '-'].include?(rune)
        first = true
        false
      else
        false
      end
    end

    runes.join('')
  end
end
